module/documents/item.mjs

```diff
@@ -392,6 +392,17 @@ export default class Item5e extends SystemDocumentMixin(Item) {
     if ( !finalSC ) return null;
     finalSC.levels = this.isEmbedded ? (this.system.levels ?? this.class?.system.levels) : null;
 
+    if (spellcasting.needToPrepare && this.isEmbedded
+      && CONFIG.DND5E.spellcastingTypes.leveled.progression[finalSC.progression]) {
+      finalSC.spellPreparationLimit = Math.max(
+        1,
+        this.parent.system.abilities[finalSC.ability].mod
+        + Math.ceil(
+          this.system.levels / (CONFIG.DND5E.spellcastingTypes.leveled.progression[finalSC.progression].divisor || 1)
+        )
+      );
+    }
+
```

Comment:
This preparation would probably be better handled within the `prepareFinalData` method on class & subclass items. We could add a static method to `SpellcastingField` to perform the calculation and then call that from the class & subclass data models:

Added to `ClassData` and `SubclassData` after the "Data Migrations" section:
```javascript
  /* -------------------------------------------- */
  /*  Data Preparation                            */
  /* -------------------------------------------- */

  /** @inheritDoc */
  prepareFinalData() {
    SpellcastingField.calculatePreparationLimit(this.spellcasting, this.parent.getRollData({ deterministic: true });
  }
```

Added to `SpellcastingField`:
```javascript
  /**
   * Calculate total preparation limit.
   * @param {object} spellcasting  Spellcasting data for this object.
   * @param {object} rollData      Roll data for the actor being prepared.
   */
  static calculatePreparationLimit(spellcasting, rollData) {
    // Calculate preparation limit here using `simplifyBonus` utility
  }
```

```suggestion
```